hyperspy/models/eelsmodel.py

```diff
@@ -794,7 +794,7 @@ def enable_free_onset_energy(self, edges_list=None):
             edges_list = [self._get_component(x) for x in edges_list]
         for edge in edges_list:
             if edge.isbackground is False:
-                edge.free_onset_energy = True
```

Comment:
Are you sure that `free_onset_energy` is disused? It is used in [`smart_fit`.](https://github.com/hyperspy/hyperspy/blob/1ffb3fab33e607045a37f30c1463350b72617e10/hyperspy/models/eelsmodel.py#L540-L545)

It should be possible to deprecate it and raise a warning when it is used.

Reply:
Initially it seemed like it was some leftover functionality, but now I'm not sure how best to handle it.

I'm guessing it is a bit confusing having both `free_onset_energy` and `onset_energy.free`. In addition, the `onset_energy.free` follows the "parameter" paradime, which is consistent with how the model fitting is done elsewhere?

I guess the `enable_free_onset_energy` function can set both `edge.free_onset_energy = True` and `edge.onset_energy.free = True`.

Then we can add a deprecation warning for `free_onset_energy`, and change `_fit_edge` from

```python
if edge.free_onset_energy is True:
    edge.onset_energy.free = True
    self.fit(**kwargs)
    edge.onset_energy.free = False
```

to 

```python
if edge.onset_energy.free is True:
    self.fit(**kwargs)
    edge.onset_energy.free = False
```